src/react/account/AccountUpdateUser.tsx

```diff
@@ -0,0 +1,191 @@
+          queryClient.setQueryData(['listIAMUsers', accountName], (old: any) => {
+            if (old) {
+              const pages: any = old.pages;
+              pages.some((page: any) => {
+                const users = page.Users;
+                const index = users.findIndex(
+                  (user: any) => user.UserName === oldUserName,
+                );
+                return index !== -1 && (users[index].UserName = newUserName);
+              });
+              return {
+                ...old,
+                pages,
+              };
+            }
+          });
+        })
+        .catch((err) => {
+          dispatch(handleErrorMessage(`${err}`, 'byModal'));
+        })
+        .finally(() => {
+          dispatch(networkEnd());
+        });
+    },
+    {
+      onSuccess: () => {
+        history.goBack();
+      },
+      onError: () => {
+        const str = 'An error occurred during the user delete.';
+        dispatch(handleErrorMessage(str, 'byModal'));
+      },
+    },
+  );
+
+  /**
+   * This part has to be handle
+   */
+  const hasError = useSelector(
+    (state: AppState) =>
+      !!state.uiErrors.errorMsg && state.uiErrors.errorType === 'byComponent',
+  );
+  const errorMessage = useSelector(
+    (state: AppState) => state.uiErrors.errorMsg,
+  );
+  const loading = useSelector(
+    (state: AppState) => state.networkActivity.counter > 0,
+  );
+
+  const onSubmit = ({ name }: {name: string}) => {
+    clearServerError();
+    updateUserMutation.mutate(name);
+  };
+
+  const handleCancel = (e: any) => {
+    if (e) {
+      e.preventDefault();
+    }
+
+    clearServerError();
+    history.goBack();
+  };
+
+  const clearServerError = () => {
+    if (hasError) {
+      dispatch(clearError());
+    }
+  };
+
+  const formRef = useRef(null);
+  useOutsideClick(formRef, clearServerError);
+  return (
+    <FormContainer>
+      <F.Form
+        autoComplete="off"
+        ref={formRef}
+        onSubmit={handleSubmit(onSubmit)}
+      >
+        <F.Title> Edit a User</F.Title>
+
+        <F.Fieldset style={{ width: '100%', flexDirection: 'row' }}>
+          <F.Label
+            tooltipMessages={[(<div key={IAMUserName} style={{ textAlign: 'start' }}>
+              <div>The ARN and the (friendly) name for the user will be edited, but the unique ID remains the same.
+              </div>
+              <br />
+              <div>The User stays in the same Groups, under its new name.</div>
+              <br />
+              <div>Policies:</div>
+              <div>- Any Policies attached to the user stays with this user, under its new name.</div>
+              <div>- Any Role (Trust) Policies that refer to the User as a Principal are automatically updated with the
+                new name.
+              </div>
+              <div>- Any Policies that refer to the User as a Resource are not updated, you have to do it manually.
+              </div>
+            </div>).toString()]
```

Comment:
I believe you add `.toString()` to handle the Typescript errors.
However you now have this : 
![](https://puu.sh/INvYG/6007fc1251.png)

The correct way to fix Typescript Errors should be check what type is missing.
In this case, it's from the `LabelProps` (src/react/ui-elements/FormLayout.tsx)

```diff
type LabelProps = {
  children: JSX.Element | string;
-  tooltipMessages?: Array<string>;
+  tooltipMessages?: Array<JSX.Element | string>;
  tooltipWidth?: string;
  style?: CSSStyleSheet;
};
```